### ISSUE TITLE
Problems building singularity image: Authentication error.

### ISSUE THREAD

**Issue by orzechoj** [nf-core/chipseq#8]
I'm having problems building the singularity image for the ChIP-seq pipeline. I get the same error both on my macbook and running on Uppmax/Rackham. 

```
$ singularity pull --name nfcore-chipseq.img docker://nf-core/ChIPseq 
WARNING: pull for Docker Hub is not guaranteed to produce the
WARNING: same image on repeated pull. Use Singularity Registry
WARNING: (shub://) to pull exactly equivalent images.
ERROR Authentication error, exiting.
Cleaning up...
ERROR: pulling container failed!

$ singularity pull --name nfcore-chipseq.img shub://nf-core/ChIPseq 
ERROR Cannot find image. Is your capitalization correct?
```

Looking at this page, perhaps the name should be _docker://nfcore/ChIPseq_ (without the "-"):  https://hub.docker.com/r/nfcore/chipseq/ But this gives the same results as above.

Building the old singularity image _docker://scilifelab/ngi-chipseq_ still works ok.

**Comment by ewels** [comment]
Have you tried `nfcore-chipseq`? (All lower caps). I think this should work. Annoyingly, docker assumes that if it can’t find a repo then it must be private and that it’s an authentication problem. So bit of a misleading error.

Images aren’t pushed to shub currently. Need to fix the docs here.

Phil

**Comment by orzechoj** [comment]
Well I got a different error message for `nfcore-chipseq`:

```
$ singularity pull --name nfcore-chipseq.img docker://nfcore-chipseq
WARNING: pull for Docker Hub is not guaranteed to produce the
WARNING: same image on repeated pull. Use Singularity Registry
WARNING: (shub://) to pull exactly equivalent images.
Docker image path: index.docker.io/library/nfcore-chipseq:latest
ERROR UNAUTHORIZED: authentication required
ERROR Check existence, naming, and permissions
Cleaning up...
ERROR: pulling container failed!
```

**Comment by ewels** [comment]
Sorry, should't reply on my phone! It's a slash, not a hyphen. No hyphens at all, all lower case:

```
$ singularity pull --name nfcore-chipseq.img docker://nf-core/chipseq
```

You can tell from the docker command box on the right on the [image page](https://hub.docker.com/r/nfcore/chipseq/) which has the image path.

Phil

**Comment by orzechoj** [comment]
Hmm, I'm still having the same problem 🤔

```
$ singularity pull --name nfcore-chipseq.img docker://nf-core/chipseq
WARNING: pull for Docker Hub is not guaranteed to produce the
WARNING: same image on repeated pull. Use Singularity Registry
WARNING: (shub://) to pull exactly equivalent images.
Docker image path: index.docker.io/nf-core/chipseq:latest
ERROR UNAUTHORIZED: authentication required
ERROR Check existence

```

**Comment by ewels** [comment]
You put a hyphen in again! It's `nfcore/chipseq` not `nf-core/chipseq` 😆 However, I confess that I did not test the command that I wrote above when I wrote it - I just tried it now and I get a similar error.

On closer inspection the reason for this becomes (embarassingly) clear. [There are no image builds](https://hub.docker.com/r/nfcore/chipseq/builds/). So there's nothing to pull.

The last one failed during the R installation step. I remember now that I did see this I think, but we're currently rebuilding the entire container, so I figured that it wasn't worth looking into it any more.

Ok, so in summary:

1. Sorry for making you test so many commands! 😆 
2. We've started making a `nf-core` python helper tool, and I will soon add a subcommand that will automatically handle all of this crap by looking at the pipeline code. So you will never have to go through this again.
3. We will [finish work on the new container](https://github.com/nf-core/chipseq/issues/2) soon I hope, at which point the `nfcore/chipseq` container will exist!
4. For now, running with the `scilifelab/ngi-chipseq` container should be fine. All of the required software is there, though it doesn't reach the level of reproducibility that we're aiming for (but this repo doesn't have a stable release yet, so we'll get there).

Again, apologies for the half-responses over the past few days (I've been travelling around teaching on various courses). Thanks for testing! I'll close this now that we have an explanation I think.

Phil
